RecoHGCal/TICL/plugins/PatternRecognitionbyCA.cc

```diff
@@ -77,7 +84,7 @@ void PatternRecognitionbyCA::makeTracksters(const PatternRecognitionAlgoBase::In
                                     min_cos_pointing_,
                                     etaLimitIncreaseWindow_,
                                     missing_layers_,
-                                    rhtools_.lastLayerFH(),
+                                    rhtools_.lastLayer(type),
```

Comment:
The HGCAL geometry is irregular.
As matter of fact the bhLastLayer_ and fhLastLayer_ point all to the last layers.

See slide 21 for the cartoon and pointers 
https://indico.cern.ch/event/918386/#8-update-on-hfnose-reconstruct
This is something I discussed in the past with @rovere , because I was confused as well.

Reply:
Uhm... I believe you, but from the definitions in RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc this is far from being obvious. 

If this is something that is supposed to last, then probably the definitions in RecHitTools should get updated, or at least made more understandable, If there is still the possibility that the two numbers sometime can become different, then this must be taken into account in the code which accesses them.

Reply:
This is related to #26225